core/store/src/lib.rs

```diff
@@ -516,6 +516,43 @@ impl CompiledContractCache for StoreCompiledContractCache {
     }
 }
 
+/// Constant for the maximum number of heights back from the current height for which
+/// blockhashes will be served. (See usage in `BlockHashProvider` impl below).
+#[cfg(feature = "protocol_feature_block_hash_host_fn")]
+const MAX_HEIGHT_HISTORY: u64 = 256;
```

Comment:
I think it is probably better to make this an associated constant of `BlockHashProvider`

Reply:
Unfortunately this does not work with the trait object approach we are currently using

```
error[E0038]: the trait `BlockHashProvider` cannot be made into an object
    --> core/primitives/src/utils.rs:373:21
     |
373  | impl fmt::Debug for dyn crate::types::BlockHashProvider {
     |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `BlockHashProvider` cannot be made into an object
     |
     = help: consider moving `MAX_HEIGHT_HISTORY` to another trait
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
    --> core/primitives/src/types.rs:1000:11
     |
999  | pub trait BlockHashProvider {
     |           ----------------- this trait cannot be made into an object...
1000 |     const MAX_HEIGHT_HISTORY: u64 = 256;
     |           ^^^^^^^^^^^^^^^^^^ ...because it contains this associated `const`
```

Speaking of which, do you have comments on https://github.com/near/nearcore/pull/4273#issuecomment-832750663 ?